src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp

```diff
@@ -958,7 +959,7 @@ namespace OpenMS
     }
 
     // store all data manipulation performed on the features of the transition group
-    transition_group = transition_group_detection;
+    transition_group = std::move(transition_group_detection);
```

Comment:
Question: Why is std::move() here necessary? 
If I understood correctly, the move assignment operator of MRMTransitionGroup should be used here automatically? Why would a copy be performed here instead? 

note:
 MRMTransitionGroupType transition_group_detection;
 typedef MRMTransitionGroup< MSChromatogram, TransitionType> MRMTransitionGroupType;

Reply:
its not necessary, but its potentially faster. 

## Scenario 1:

```
transition_group = transition_group_detection;
```

everything gets copied, we now have 2 objects that contain all the data, all chromatograms etc. So it means we need to spend time to copy everything over to `transition_group`

## Scenario 2:

```
transition_group = std::move(transition_group_detection);
```
we `move` all the data out of `transition_group_detection` into `transition_group` which for larger data structures mostly involves copying a ptr to a new location. That is much much faster than copying the data and keeping it twice. Since we are not going to use `transition_group_detection` any more, its ok if its empty and does not containt he data any more